src/signal.rs

```diff
@@ -14,35 +13,32 @@ pub enum SignalTo {
 /// Signals from OS/user.
 #[cfg(unix)]
 pub fn signals() -> impl Stream<Item = SignalTo> {
-    use tokio_signal::unix::{Signal, SIGHUP, SIGINT, SIGQUIT, SIGTERM};
+    use tokio::signal::unix::{signal, SignalKind};
 
-    let sigint = Signal::new(SIGINT).flatten_stream();
-    let sigterm = Signal::new(SIGTERM).flatten_stream();
-    let sigquit = Signal::new(SIGQUIT).flatten_stream();
-    let sighup = Signal::new(SIGHUP).flatten_stream();
+    let mut sigint = signal(SignalKind::interrupt()).expect("Signal handlers should not panic.");
+    let mut sigterm = signal(SignalKind::terminate()).expect("Signal handlers should not panic.");
+    let mut sigquit = signal(SignalKind::quit()).expect("Signal handlers should not panic.");
+    let mut sighup = signal(SignalKind::hangup()).expect("Signal handlers should not panic.");
 
-    let signals = sigint.select(sigterm.select(sigquit.select(sighup)));
-
-    signals
-        .map(|signal| match signal {
-            SIGHUP => SignalTo::Reload,
-            SIGINT | SIGTERM => SignalTo::Shutdown,
-            SIGQUIT => SignalTo::Quit,
-            _ => unreachable!(),
-        })
-        .compat()
-        .into_stream()
-        .map(|result| result.expect("Neither stream errors"))
+    async_stream::stream! {
+        let signal = tokio::select! {
+            _ = sigint.recv() => SignalTo::Shutdown,
+            _ = sigterm.recv() => SignalTo::Shutdown,
+            _ = sigquit.recv() => SignalTo::Quit,
+            _ = sighup.recv() => SignalTo::Reload,
+        };
+        yield signal;
+    }
 }
 
 /// Signals from OS/user.
 #[cfg(windows)]
 pub fn signals() -> impl Stream<Item = SignalTo> {
-    let ctrl_c = tokio_signal::ctrl_c().flatten_stream();
+    use futures::future::FutureExt;
+    let ctrl_c = tokio::signal::ctrl_c();
 
-    ctrl_c
-        .map(|_| SignalTo::Shutdown)
-        .compat()
-        .into_stream()
-        .map(|result| result.expect("Shouldn't error"))
+    async_stream::stream! {
+        let signal = ctrl_c.map(|_| SignalTo::Shutdown).await;
```

Comment:
`ctrl_c` is still a future here, not a stream, so I'm pretty sure this is the same as before. That said, if we don't need to handle it more than once here then this is fine. We handle it twice on linux because the first is "shutdown nicely please" and the second is "shutdown immediately".

Reply:
I tried 

```rust
    async_stream::stream! {
        let mut signal;
        loop {
            tokio::signal::ctrl_c().map(|_| SignalTo::Shutdown).await;
        }
        yield signal;
    }
}
```

But that left me with a binary that didn't handle ctrl  C.

Reply:
Yeah I thought I actually wrote a loop...